lib/metasploit/framework/compiler/mingw.rb

```diff
@@ -59,15 +60,54 @@ def build_cmd(src)
           cmd
         end
 
-        def cleanup_files
-          src_file = "#{self.file_name}.c"
-          exe_file = "#{self.file_name}.exe"
 
+        def compile_cpp(src)
+          cmd = build_cpp_cmd(src)
+          print("#{cmd}\n") if show_compile_cmd
+          stdin_err, status = Open3.capture2e(cmd)
+          stdin_err
+        end
+
+        def compile_cpp_files(files)
+          cmd = build_cpp_files_cmd(files)
+          print("#{cmd}\n") if show_compile_cmd
+          stdin_err, status = Open3.capture2e(cmd)
+          stdin_err
+        end
+
+        def build_cpp_cmd(src)
+          src_file = Tempfile.create(['scexec', ".cpp"])
+          File.write(src_file.path, src)
+          build_cpp_files(src_file.path)
```

Comment:
Was this a typo? There isn't any function by this name within the codebase that I can see or within this edit. I assume you meant to call `build_cpp_files_cmd` but I would note that it appears to be taking in an array, instead of the string you are currently passing in.

Reply:
oh lord, yes, thank you.